### ISSUE TITLE
Code question

### ISSUE THREAD

**Issue by TuffyY** [aj1365/PolSARFormer#4]
Thanks for your very nice code.I met a problem with your code:

According to my calculations, the shape of the `patchesData` returned from the `ImageCube` function should be `(H*W, 12, 12, 9)`. However, I noticed that you are reshaping `x` to have a shape of `(X1.shape[0], windowSize, windowSize, 12, 1)` based on the setting of `windowSize=12` in your paper, which translates to `(H, 12, 12, 12, 1)`. This shape doesn't match my calculated shape of `H*W, 12, 12, 9`. Therefore, I have the following doubts:

1. Should the first dimension of the reshape operation be `H*W`? If my understanding is incorrect, I would like to know how you performed this step.

2. For the third dimension of my data, which represents the scattering dimension and has a size of 9, I am curious about the fourth dimension in your reshape operation, which is the last `12`. Is this `12` related to the scattering dimension in your dataset? If not, I would like to understand the significance of this `12`.

3. After running the code, I noticed that our data shapes do not seem to match. Could you please confirm the dimensionality of the scattering data in your San Francisco dataset? This information would be extremely helpful for my research. 
Thank you very much.

**Comment by aj1365** [comment]
Hi, It would be (N, 12, 12, 12, 1). N is the number of ground truth data (training data + test data). 
Look at the place, I have reshaped the data. 
I am just creating a 3D cube out of the ground truth data.

**Comment by aj1365** [comment]
Look at input shape in PolSARFormer (12, 12, 12, 1).

**Comment by TuffyY** [comment]
So in this Code:X1 = X1.reshape((X1.shape[0], windowSize, windowSize, 12, 1)),Is this 12  related to the scattering dimension? I know that (windowSize,windowSize)is shape of the cube.But I do not know what is this 12 means. 
Thank u very much!

**Comment by aj1365** [comment]
(N, height, width, channels, 1)= (N, 12, 12, 12, 1)
X1.shape[0] or N is the number of ground truth samples.

**Comment by rongtongxueya** [comment]
I have a question about this place? Does this mean that a patch size in the windowSize range is a tag? Instead of pixel-scale labels? Instead of having a label for each pixel, there is a windowSize size for each label class?

**Comment by aj1365** [comment]
This is how CNN models work. There is a label for an image patch.
